clients/unshaded/src/main/java/tachyon/client/BlockOutStream.java

```diff
@@ -61,7 +61,10 @@ public static BlockOutStream get(TachyonFile tachyonFile, WriteType opType, int
 
     if (tachyonFile.mTachyonFS.hasLocalWorker()
         && tachyonConf.getBoolean(Constants.USER_ENABLE_LOCAL_WRITE,
-            Constants.DEFAULT_USER_ENABLE_LOCAL_WRITE)) {
+            Constants.DEFAULT_USER_ENABLE_LOCAL_WRITE)
+            && tachyonConf.get(
```

Comment:
I need this condition because the master will have returned a worker earlier. However, the worker returned can not be local even if we have one local worker. In this case it will try to write with a LocalBlockOutStream and fail because the worker is not local and need to be a RemoteBlockOutStream.